### PR TITLE
Bump TypeScript version to allow for optional chaining.

### PR DESCRIPTION
**Pull request by mbleigh**
New TypeScript, who dis?

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,102 +1,99 @@
 module.exports = {
```
Comment (mbleigh) [inline .eslintrc.js]: Prettier automatically dropped all the quotes which seems, preferable?

File: src/emulator/types.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export function isJavaEmulator(value: string): value is JavaEmulators {
 }
 
 export function isEmulator(value: string): value is Emulators {
-  return Object.values(Emulators).indexOf(value) >= 0;
+  return Object.values(Emulators).indexOf(value as Emulators) >= 0;
```
Comment (mbleigh) [inline src/emulator/types.ts]: This one was weird and seems wrong but it's the only way I could get it to compile. TS seems like it should be smarter here.

File: src/init/features/project.ts [hunk 3]
```diff
@@ -104,7 +104,7 @@ async function projectChoicePrompt(options: any): Promise<FirebaseProjectMetadat
  * @param config Configuration for the project.
  * @param options Command line options.
  */
-export async function doSetup(setup: any, config: Config, options: any): Promise<void> {
+export async function doSetup(setup: any, config: any, options: any): Promise<void> {
```
Comment (mbleigh) [inline src/init/features/project.ts]: I tried really hard not to make this an `any` but it wasn't gonna happen.

File: .eslintrc.js (hunk 1 above)
Comment (bkendall) [reply .eslintrc.js]: sure

**Pull request merged**